src/LossFunctions.jl

```diff
@@ -1,45 +1,47 @@
 using Random: randperm
 
 # Sum of square error between two arrays
-function SSE(x::Array{Float32}, y::Array{Float32})::Float32
+function SSE(x::AbstractArray{T}, y::AbstractArray{T})::T where {T<:AbstractFloat}
```

Comment:
`AbstractFloat` seems a little tighter than necessary for a lot of these. If you want to put a bound on it, I think `Real` might be the right level. For example, rational numbers should be fine.